themes/bootstrap5/scss/components/offcanvas.scss

```
@@ -29,70 +29,72 @@ $offcanvas-offset: 80vw !default;  // Width of open menu
-    cursor: pointer;
-  }
-  .vufind-offcanvas .vufind-offcanvas-overlay { display: block; }
+    body.vufind-offcanvas {
+      overflow-x: hidden; /* Prevent scroll on narrow devices */
+      transition: margin .5s;
+    }
 
-  .vufind-offcanvas-left {
-    .sidebar { left: -$offcanvas-offset; }
-    .vufind-offcanvas-overlay { right: -100%; }
-    .vufind-offcanvas-toggle { left: 0; }
-  }
-  .vufind-offcanvas-right {
-    .sidebar { right: -$offcanvas-offset; }
-    .vufind-offcanvas-overlay { left: -100%; }
-    .vufind-offcanvas-toggle { right: 0; }
-  }
-  .vufind-offcanvas .sidebar
-  {
-    position: fixed;
-    z-index: 15;
-    height: 100%;
-    top: 0;
-    width: $offcanvas-offset;
-    padding: 1rem;
-    overflow-y: auto;
-    background-color: #fff;
-    transition: left .5s, right .5s;
+    .vufind-offcanvas-overlay {
+      display: none;
+      position: fixed;
+      top: 0;
+      width: 100%;
+      height: 100%;
+      background-color: rgba(0,0,0,.3);
+      z-index: 14;
+      opacity: 0;
+      transition: opacity .5s;
+      cursor: pointer;
+    }
+    .vufind-offcanvas .vufind-offcanvas-overlay { display: block; }
 
-    .search-sort { padding-bottom: 1rem; }
-  }
-  .vufind-offcanvas.active {
-    overflow-y: hidden;
-    &.vufind-offcanvas-left {
-      .sidebar { left: 0; }
-      .vufind-offcanvas-overlay { right: -$offcanvas-offset; }
-      .vufind-offcanvas-toggle { left: $offcanvas-offset; }
+    .vufind-offcanvas-left {
+      .sidebar { left: -$offcanvas-offset; }
+      .vufind-offcanvas-overlay { right: -100%; }
+      .vufind-offcanvas-toggle { left: 0; }
     }
-    &.vufind-offcanvas-right {
-      .sidebar { right: 0; }
-      .vufind-offcanvas-overlay { left: -$offcanvas-offset; }
-      .vufind-offcanvas-toggle { right: $offcanvas-offset; }
+    .vufind-offcanvas-right {
+      .sidebar { right: -$offcanvas-offset; }
+      .vufind-offcanvas-overlay { left: -100%; }
+      .vufind-offcanvas-toggle { right: 0; }
     }
-    .vufind-offcanvas-overlay {
-      opacity: 1;
-      transition: opacity .5s;
+    .vufind-offcanvas .sidebar
+    {
+      position: fixed;
+      z-index: 15;
+      height: 100%;
+      top: 0;
+      width: $offcanvas-offset;
+      padding: 1rem;
+      overflow-y: auto;
+      background-color: #fff;
+      transition: left .5s, right .5s;
+
+      .search-sort { padding-bottom: 1rem; }
+    }
+    .vufind-offcanvas.active {
+      overflow-y: hidden;
+      &.vufind-offcanvas-left {
+        .sidebar { left: 0; }
+        .vufind-offcanvas-overlay { right: -$offcanvas-offset; }
+        .vufind-offcanvas-toggle { left: $offcanvas-offset; }
+      }
+      &.vufind-offcanvas-right {
+        .sidebar { right: 0; }
+        .vufind-offcanvas-overlay { left: -$offcanvas-offset; }
+        .vufind-offcanvas-toggle { right: $offcanvas-offset; }
+      }
+      .vufind-offcanvas-overlay {
+        opacity: 1;
+        transition: opacity .5s;
+      }
     }
   }
-}
+}
```

End of file line-break?

```suggestion
}

```